airbyte-integrations/connectors/source-instatus/source_instatus/components.py

```diff
@@ -0,0 +1,114 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from dataclasses import InitVar, dataclass
+from typing import Any, Iterable, List, Mapping, Optional
+
+import dpath.util
+from airbyte_cdk.models import AirbyteMessage, SyncMode, Type
+from airbyte_cdk.sources.declarative.stream_slicers.substream_slicer import ParentStreamConfig, SubstreamSlicer
+from airbyte_cdk.sources.declarative.transformations import AddFields
+from airbyte_cdk.sources.declarative.types import Config, Record, StreamSlice, StreamState
+
+
+@dataclass
+class ListAddFields(AddFields):
+    """
+    ListAddFields uses to transform record by adding an ids from list object field to one list.
```

Comment:
For both of these custom components can you please include a quick summary in the PR description and in these comments why the existing `AddFields` component and the `UpdatesSubstreamSlicer` were not suitable for reuse by the Instatus API?

Reply:
Added two custom components ListAddFields and UpdatesSubstreamSlicer. 
This was done for the maintenance_updates_stream and incident_updates_stream to work correctly. 
The work algorithm is as follows, in the parent streams maintenance and incident is a list of all update objects. The ListAddFields component selects their IDs from these objects and adds them to the new field "updates_ids", which is a list. 
Next, UpdatesSubstreamSlicer creates a stream of slices, additionally iterating over the list of updates_ids and adding them to the slice with the correct parent id.
That is, ListAddFields solves the problem that we do not have a separate list of all updates_ids at the beginning and cannot extract them from the list of update objects. And UpdatesSubstreamSlicer has an additional iteration over the list of updates and creates a correct stream slice.